src/main/java/com/atlassian/db/replica/api/strategy/PropagateReplicaFailure.java

```diff
@@ -0,0 +1,17 @@
+package com.atlassian.db.replica.api.strategy;
+
+import com.atlassian.db.replica.api.SqlCall;
+import com.atlassian.db.replica.spi.ReplicaFailureStrategy;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+
+/**
+ * Propagates replica failure.
+ */
+public class PropagateReplicaFailure implements ReplicaFailureStrategy {
+    @Override
+    public Connection onFailure(SQLException exception, SqlCall<Connection> mainConnection) throws SQLException {
+        throw exception;
```

Comment:
what if exception would be other type than `SQLException`

Reply:
`SQLException` is part of the `Connection` API and `ConnectionProvider` SPI.  A library customer provides ConnectionProvider implementation. It's up to the customer to ensure the contract.